Assets/Scripts/Framework/Gameplay/LaneSystem/Lane.cs

```diff
@@ -0,0 +1,147 @@
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections.Generic;
+
+using ProefExamen.Framework.Gameplay.Values;
+using ProefExamen.Framework.Utils.Libraries.LaneUtils;
+
+namespace ProefExamen.Framework.Gameplay.LaneSystem
+{
+    /// <summary>
+    /// A class that manages a lane.
+    /// </summary>
+    [RequireComponent(typeof(Button))]
+    public class Lane : MonoBehaviour
+    {
+        [SerializeField]
+        private int _laneID = -1;
+
+        [Header("References")]
+        [SerializeField]
+        private Button _button;
+
+        [SerializeField]
+        private List<Note> _notes = new();
+
+        [Header("Lerp positions")]
+        [SerializeField]
+        private Vector3 _initialNotePosition;
+
+        [SerializeField]
+        private Vector3 _targetNotePosition;
+
+        /// <summary>
+        /// Gets the registered Notes of this lane.
+        /// </summary>
+        public List<Note> Notes => _notes;
+
+        /// <summary>
+        /// Gets the private button component of the Lane.
+        /// </summary>
+        public Button Button => _button;
+
+        private void Awake()
+        {
+            if(_button == null)
+                _button = gameObject.GetComponent<Button>();
+        }
+
+        private void Start()
+        {
+            if (_button)
```

Comment:
Use `_button == null` and invert the if.
```suggestion
            if (_button == null)
```